includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign-usage-reports.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Service Provider: Active_Campaign Usage Reports
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Usage reports for Active_Campaign.
+ */
+class Newspack_Newsletters_Active_Campaign_Usage_Reports {
+	/**
+	 * Retrieves the main Active_Campaign instance
+	 *
+	 * @return Newspack_Newsletters_Active_Campaign
+	 */
+	private static function get_ac_instance() {
+		return Newspack_Newsletters_Active_Campaign::instance();
+	}
+
+	/**
+	 * Get all contacts.
+	 *
+	 * @param int   $status Status of contacts to get.
+	 * @param date  $created_after Date after which contacts were created.
+	 * @param array $contacts Array of contacts.
+	 */
+	private static function get_all_contacts( $status = 1, $created_after = null, $contacts = [] ) {
```

Comment:
I just thought of a more efficient way to do this.

Instead of having to do an unpredictable number of requests and loop though all contacts, we could simply ask for one contact "created_after" (or "updated_after") and look at the total number in the response for today.

Then we do the same for 1 day before, and subtract the number of today to get the number of yesterday... and so on for evey day in the past.

This would ensure only one request per day and a small response ( we only need one contact ).

What do you think?

This is what I'm planning to do for Campaign Monitor (1205653098588469-as-1205653098588484)

Reply:
Huh, that sounds like a smart optimisation! In the interest of time, I'd prefer to consider it only if we run into issues with the current implementation (let's not optimize prematurely).